bot/src/main/java/com/dongtronic/diabot/platforms/discord/listeners/QuoteListener.kt

```diff
@@ -36,55 +32,58 @@ class QuoteListener(private val client: CommandClient) : CoroutineEventListener
         }
     }
 
+    @Suppress("CyclomaticComplexMethod", "ReturnCount")
     private suspend fun onMessageReactionAdd(event: MessageReactionAddEvent) {
-        if (event.retrieveUser().await().isBot) return
         if (!event.isFromGuild) return
-        if (event.reaction.emoji !is UnicodeEmoji) return
-        if (event.reaction.emoji != speechEmoji) return
+        if (event.reaction.emoji !is UnicodeEmoji || event.reaction.emoji != speechEmoji) return
```

Comment:
Why do we now have the `isFromGuild` check on a separate line, but these two together with `||`?

Reply:
I was trying to reduce the number of separate `if` statements, and I felt like these two were related enough to merge them together. I can separate them if you'd prefer the old format though.